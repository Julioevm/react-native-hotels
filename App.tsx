import React, {useState} from 'react';
import {StatusBar, useColorScheme} from 'react-native';
import {NavigationContainer} from '@react-navigation/native';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import ListScreen from './screens/ListScreen';
import DetailsScreen from './screens/DetailsScreen';
import {RootStackParamList} from './types';
import {GestureHandlerRootView} from 'react-native-gesture-handler';
import SortScreen from './screens/SortScreen';
import FilterContext, {Filters, defaultFilter} from './context/FilterContext';
import {Theme} from './Theme';

const App = () => {
  const isDarkMode = useColorScheme() === 'dark';
  const [filters, setFilters] = useState<Filters>(defaultFilter);
  const Stack = createNativeStackNavigator<RootStackParamList>();

  return (
    // eslint-disable-next-line react-native/no-inline-styles
    <GestureHandlerRootView style={{flex: 1}}>
      <FilterContext.Provider value={{filters, setFilters}}>
        <NavigationContainer>
          <StatusBar
            barStyle={isDarkMode ? 'light-content' : 'dark-content'}
            backgroundColor={Theme.colors.secondary}
          />
          <Stack.Navigator
            screenOptions={{
              headerStyle: {
                backgroundColor: Theme.colors.secondary,
              },
              headerTintColor: '#fff',
            }}>
            <Stack.Screen
              name="Hotels"
              component={ListScreen}
              options={{
                headerShown: false,
              }}
            />
            <Stack.Screen
              name="Details"
              component={DetailsScreen}
              options={({route}) => ({
                title: route.params.hotel.name,
                headerTitleStyle: {
                  fontSize: 16,
                  fontWeight: 'bold',
                },
              })}
            />
            <Stack.Screen
              name="Sort"
              component={SortScreen}
              options={{
                headerShown: false,
                presentation: 'modal',
              }}
            />
          </Stack.Navigator>
        </NavigationContainer>
      </FilterContext.Provider>
    </GestureHandlerRootView>
  );
};

export default App;
