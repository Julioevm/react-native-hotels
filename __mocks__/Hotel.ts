export const Hotel = {
  id: '1',
  name: 'Hotel',
  stars: 4,
  userRating: 4.5,
  price: 100,
  currency: 'EUR',
  location: {
    latitude: 0,
    longitude: 0,
    city: 'Madrid',
    country: 'Spain',
    address: 'Calle de la Princesa, 1',
  },
  checkIn: {from: '2020-01-01', to: '2020-01-02'},
  checkOut: {from: '2020-01-01', to: '2020-01-02'},
  contact: {
    phoneNumber: '123456789',
    email: 'test@mail.com',
  },
  gallery: ['https://picsum.photos/200/300'],
};
