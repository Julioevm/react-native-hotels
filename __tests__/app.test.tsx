import React from 'react';
import {render, cleanup, waitFor} from '@testing-library/react-native';
import App from '../App';
import 'jest-fetch-mock';
import {Hotels} from '../__mocks__/Hotel';
jest.mock('../components/common/Carrousel.tsx', () => 'Carrousel');

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({
    navigate: jest.fn(),
  }),
}));

describe('App', () => {
  afterEach(cleanup);

  it('renders correctly', async () => {
    fetchMock.mockOnce(JSON.stringify(Hotels));
    const {toJSON} = render(<App />);
    waitFor(() => {
      expect(toJSON()).toMatchSnapshot();
    });
  });

  it('should display the no data message', () => {
    fetchMock.mockOnce(JSON.stringify([]));
    const app = render(<App />);
    expect(app.findByText('no data found :(')).toBeTruthy();
  });
});
