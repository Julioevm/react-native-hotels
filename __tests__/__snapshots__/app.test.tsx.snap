// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        collapsable={false}
        forwardedRef={[Function]}
        gestureResponseDistance={
          Object {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        nativeBackButtonDismissalEnabled={false}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onHeaderBackButtonClicked={[Function]}
        onNativeDismissCancelled={[Function]}
        onTransitionProgress={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        stackPresentation="push"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        swipeDirection="horizontal"
      >
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backgroundColor="tomato"
          color="#fff"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={false}
          hideBackButton={false}
          largeTitleHideShadow={false}
          title="Hotels"
          titleColor="#fff"
          topInsetEnabled={false}
          translucent={false}
        />
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            Object {
              "flex": 1,
              "flexDirection": "column-reverse",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "lavenderblush",
                    "flex": 1,
                  }
                }
              >
                <RCTScrollView
                  getItem={[Function]}
                  getItemCount={[Function]}
                  keyExtractor={[Function]}
                  onContentSizeChange={[Function]}
                  onLayout={[Function]}
                  onMomentumScrollBegin={[Function]}
                  onMomentumScrollEnd={[Function]}
                  onScroll={[Function]}
                  onScrollBeginDrag={[Function]}
                  onScrollEndDrag={[Function]}
                  removeClippedSubviews={false}
                  renderItem={[Function]}
                  scrollEventThrottle={50}
                  stickyHeaderIndices={Array []}
                  viewabilityConfigCallbackPairs={Array []}
                >
                  <View />
                </RCTScrollView>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenStack>
  </RNCSafeAreaProvider>
</View>
`;
